### DISCUSSION TITLE
ketcher-react

### DISCUSSION THREAD

**Discussion by KatyaMarush** [epam/ketcher#1510] [Q&A]
Is there any way to get smiles after editing it for ketcher-react Editor?

**Comment by AndreiMazol** [comment]
Dear @KatyaMarush ,

sorry for delay. To get SMILES you can use the following code:

`const smiles = await ketcher.getSmiles()`

the link to  ketcher instance can be saved as a part on Editor -> onInit property: 

`onInit={(ketcher: Ketcher) => {
          ;(global as any).ketcher = ketcher
        }
}`

for example please look at https://github.com/epam/ketcher/blob/abb79b215afaea822f616360a6f6ed40ef8a4f89/example/src/App.tsx#L70

Regards,
Ketcher team
